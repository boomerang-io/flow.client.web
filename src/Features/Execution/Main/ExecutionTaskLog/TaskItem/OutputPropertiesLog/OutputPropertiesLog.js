import React from "react";
import PropTypes from "prop-types";
import { Button, ComposedModal, ModalForm } from "@boomerang-io/carbon-addons-boomerang-react";
import { ModalBody } from "@boomerang-io/carbon-addons-boomerang-react";
import PropertiesTable from "./PropertiesTable";
import styles from "./outputPropertisLog.module.scss";

function OutputPropertiesLog({ flowTaskName, flowTaskOutputs, isOutput }) {
  let arrayProps = [];
  Object.keys(flowTaskOutputs).forEach(
    (val, index) =>
      (arrayProps = arrayProps.concat({
        id: `${val}-${index}`,
        key: val,
        value: JSON.stringify(flowTaskOutputs[val], null, 2),
      }))
  );

  return (
    <ComposedModal
      composedModalProps={{ containerClassName: styles.container, shouldCloseOnOverlayClick: true }}
      confirmModalProps={{
        title: "Are you sure?",
        children: "Your changes will not be saved",
      }}
      modalHeaderProps={{
        title: "Output Parameters",
        label: `${flowTaskName}`,
      }}
      modalTrigger={({ openModal }) => (
        <Button kind="ghost" size="small" onClick={openModal}>
          View Parameters
        </Button>
      )}
    >
      {() => (
        <ModalForm>
          <ModalBody>
            <PropertiesTable hasJsonValues={!isOutput} data={isOutput ? flowTaskOutputs : arrayProps} />
          </ModalBody>
        </ModalForm>
      )}
    </ComposedModal>
  );
}

OutputPropertiesLog.propTypes = {
  flowTaskName: PropTypes.string.isRequired,
  flowTaskOutputs: PropTypes.object.isRequired,
};

export default OutputPropertiesLog;
