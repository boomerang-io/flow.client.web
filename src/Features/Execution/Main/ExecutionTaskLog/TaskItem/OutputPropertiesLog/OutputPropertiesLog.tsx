import React from "react";
import ReactJson from "react-json-view";
import { Button, ComposedModal, ModalForm } from "@boomerang-io/carbon-addons-boomerang-react";
import { ModalBody, Tabs, Tab } from "@boomerang-io/carbon-addons-boomerang-react";
import PropertiesTable from "./PropertiesTable";
import styles from "./outputPropertisLog.module.scss";

type Props = {
  flowTaskName: string;
  flowTaskOutputs: {
    [key: string]: string;
  };
  isOutput?: boolean;
};

function OutputPropertiesLog({ flowTaskName, flowTaskOutputs, isOutput }: Props) {
  let arrayProps: {id: string; key: string; value: string;}[] = [];
  Object.keys(flowTaskOutputs).forEach(
    (val: string, index: number) =>
      (arrayProps = arrayProps.concat({
        id: `${val}-${index}`,
        key: val,
        value: JSON.stringify(flowTaskOutputs[val], null, 2),
      }))
  );

  return (
    <ComposedModal
      composedModalProps={{ containerClassName: styles.container, shouldCloseOnOverlayClick: true }}
      confirmModalProps={{
        title: "Are you sure?",
        children: "Your changes will not be saved",
      }}
      modalHeaderProps={{
        title: "Output Parameters",
        label: `${flowTaskName}`,
      }}
      modalTrigger={({ openModal }: { openModal: () => void }) => (
        <Button kind="ghost" size="small" onClick={openModal}>
          View Parameters
        </Button>
      )}
    >
      {() => (
        <ModalForm>
          <ModalBody>
            <Tabs>
              <Tab label="Table">
                <PropertiesTable hasJsonValues={!isOutput} data={isOutput ? flowTaskOutputs : arrayProps} />
              </Tab>
              <Tab label="JSON">
                <div className={styles.propertiesJson}>
                  <ReactJson
                    name={false}
                    src={flowTaskOutputs}
                    displayDataTypes={false}
                    //@ts-ignore
                    enableDelete={false}
                    displayObjectSize={false}
                    enableEdit={false}
                    enableAdd={false}
                  />
                </div>
              </Tab>
            </Tabs>
          </ModalBody>
        </ModalForm>
      )}
    </ComposedModal>
  );
}

export default OutputPropertiesLog;
