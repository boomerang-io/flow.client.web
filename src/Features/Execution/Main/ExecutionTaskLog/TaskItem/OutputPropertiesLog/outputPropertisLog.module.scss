@import "Styles/mixins";

.container {
  max-width: 50rem;

  :global .bx--modal-content {
    padding-right: 1rem;
  }
}
