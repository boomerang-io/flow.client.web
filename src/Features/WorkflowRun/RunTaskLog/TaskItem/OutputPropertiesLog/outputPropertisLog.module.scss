@import "Styles/mixins";

.container {
  max-width: 50rem;

  :global .cds--modal-content {
    padding-right: 1rem;
  }
}
