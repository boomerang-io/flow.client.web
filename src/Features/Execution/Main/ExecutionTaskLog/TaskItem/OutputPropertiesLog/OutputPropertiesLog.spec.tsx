import React from "react";
import OutputPropertiesLog from "./index";

const props = {
  flowTaskName: "Send Slack Message",
  flowTaskOutputs: {
    args: {
      test: "true",
    },
  },
};

describe("OutputPropertiesLog --- Snapshot", () => {
  it("Capturing Snapshot of OutputPropertiesLog", () => {
    const { baseElement } = global.rtlRender(<OutputPropertiesLog {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
