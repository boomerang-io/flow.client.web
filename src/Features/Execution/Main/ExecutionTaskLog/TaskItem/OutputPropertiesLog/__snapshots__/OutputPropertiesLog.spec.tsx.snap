// Vitest Snapshot v1

exports[`OutputPropertiesLog --- Snapshot > Capturing Snapshot of OutputPropertiesLog 1`] = `
<body
  id="app"
>
  <div>
    <button
      class="cds--btn cds--btn--sm cds--btn--ghost"
      tabindex="0"
      type="button"
    >
      View Parameters
    </button>
  </div>
  <div
    class="cds--bmrg-modal-portal"
  />
</body>
`;
